Lib/test/test_sqlite3/test_dbapi.py

```diff
@@ -887,6 +887,17 @@ def test_execute_named_param_and_sequence(self):
                     self.cu.execute(query, params)
                 self.assertEqual(cm.filename,  __file__)
 
+    def test_execute_indexed_nameless_params(self):
+        # See gh-117995: "'?1' is considered a named placeholder"
+        for query, params, expected in (
+            ("select ?1, ?2", (1, 2), (1, 2)),
+            ("select ?2, ?1", (1, 2), (2, 1)),
+        ):
+            with self.subTest(query=query, params=params):
+                cu = self.cu.execute(query, params)
```

Comment:
If this change is mainly intended to get rid of the warning, maybe we can test it to ensure it won't raise warnings like this:

```python
with warnings.catch_warnings():
    warnings.simplefilter("error", DeprecationWarning)
    ...
```

Reply:
I added a basic test which will call `warnings.warn('...', DeprecationWarning)` on my local branch and it can still pass the test:

```
test_xxx (test.test_dis.TestDisTraceback.test_xxx) ... /Users/xxxxx/Codes/cpython/Lib/test/test_dis.py:2250: DeprecationWarning: xxx
  warnings.warn('xxx', DeprecationWarning)
ok
test_xxx (test.test_dis.TestDisTracebackWithFile.test_xxx) ... ok
```

I found that we can pass `-We` flags on the Python command line, which will cause the test to fail if a warning is raised (it will convert warnings to errors), but on GHA this is not specified: https://github.com/python/cpython/blob/340a02b590681d4753eef0ff63037d0ecb512271/.github/workflows/build.yml#L409-L412

So I think `warnings.simplefilter("error", DeprecationWarning)` is still needed?

Reply:
Ah, indeed, a warning does actually not make a `python -m test test_sqlite3` run fail. Nevertheless, we don't want to add such a filter; instead we would want to _fail_ if a deprecation warning is raised.

Reply:
So, I guess we could use `warnings.catch_warnings` and assert that a warning was not raised.

Reply:
I addressed your remark in eb582a02fab3c8772962fa7a124b05496225898a.

Reply:
https://github.com/python/cpython/commit/eb582a02fab3c8772962fa7a124b05496225898a is fine, but I recommend using:

```
with warnings.catch_warnings():
    warnings.simplefilter("error", DeprecationWarning)
    cu = self.cu.execute(query, params)
```

The `catch_warnings()` will restore the filter when out of the context. See https://docs.python.org/3/library/warnings.html#temporarily-suppressing-warnings